src/services/YotiService.ts

```diff
@@ -31,309 +31,330 @@ export class YotiService {
-				const xRequestId = error.response ? error.response.headers["x-request-id"] : undefined;
-				this.logger.error({ message: "An error occurred when fetching Yoti instructions PDF", yotiErrorMessage: error.message, yotiErrorCode: error.code, messageCode: MessageCodes.FAILED_YOTI_GET_INSTRUCTIONS, xRequestId });
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error fetching Yoti instructions PDF");
-			}
-		} else {
-			this.logger.error({ message: "Missing Yoti request config ", yotiRequest });
-		}
+    	const yotiRequest = this.generateYotiRequest({
+    		method: HttpVerbsEnum.GET,
+    		endpoint: `/sessions/${sessionId}/instructions/pdf`,
+    		configResponseType: "arraybuffer",
+    		configResponseEncoding: "binary",
+    	});
+
+    	if (yotiRequest && yotiRequest.config && yotiRequest.url) {
+
+    		try {
+    			const yotiRequestConfig =  yotiRequest.config;
+    			this.logger.debug("getPdf - Yoti", { yotiRequestConfig });
+    			return (await axios.get(yotiRequest.url, yotiRequest.config)).data;
+
+    		} catch (error: any) {
+    			const xRequestId = error.response ? error.response.headers["x-request-id"] : undefined;
+    			this.logger.error({ message: "An error occurred when fetching Yoti instructions PDF", yotiErrorMessage: error.message, yotiErrorCode: error.code, messageCode: MessageCodes.FAILED_YOTI_GET_INSTRUCTIONS, xRequestId });
+    			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error fetching Yoti instructions PDF");
+    		}
+    	} else {
+    		this.logger.error({ message: "Missing Yoti request config ", yotiRequest });
+    	}
 	}
 
-	async getCompletedSessionInfo(sessionId: string): Promise<YotiCompletedSession | undefined> {
-		const yotiRequest = this.generateYotiRequest({
-			method: HttpVerbsEnum.GET,
-			endpoint: `/sessions/${sessionId}`,
-		});
-
-		try {
-			const { data } = await axios.get(yotiRequest.url, yotiRequest.config);
-
-			return data;
-		} catch (error: any) {
-			const xRequestId = error.response ? error.response.headers["x-request-id"] : undefined;
-			this.logger.error({ message: "An error occurred when fetching Yoti session", yotiErrorMessage: error.message, yotiErrorCode: error.code, messageCode: MessageCodes.FAILED_YOTI_GET_SESSION, xRequestId });
-			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error fetching Yoti Session");
-		}
+	async getCompletedSessionInfo(sessionId: string, backoffPeriodMs: number, maxRetries: number): Promise<YotiCompletedSession | undefined> {
+    	const yotiRequest = this.generateYotiRequest({
+    		method: HttpVerbsEnum.GET,
+    		endpoint: `/sessions/${sessionId}`,
+    	});
+    	let retryCount = 0;
+    	while (retryCount <= maxRetries) {
+    		this.logger.debug({
+    			message: "getCompletedSessionInfo - trying to fetch Yoti session", 
+    			yotiSessionId: sessionId,
+    			retryCount,
+    		});
+			console.log("retryCount: " + retryCount + ", backoffPeriodMs: " + backoffPeriodMs + "retries: " + maxRetries);
```

Comment:
Do we still need this log? If so could we covert to use the aws logger?

Reply:
No, not required. I had then added for debugging purpose. missed taking it off. Thanks.